server/handlers/destinations.go

```diff
@@ -172,13 +249,73 @@ func testDestinationConnection(config *storages.DestinationConfig) error {
 		fbAdapter := adapters.NewTestFacebookConversion(config.Facebook)
 
 		return fbAdapter.TestAccess()
+
 	case storages.WebHookType:
 		if err := config.WebHook.Validate(); err != nil {
 			return err
 		}
 
 		return nil
+
 	default:
-		return errors.New("unsupported destination type " + config.Type)
+		return fmt.Errorf("Unsupported destination type: '%v'", config.Type)
 	}
 }
+
+func TestProcessing(config *storages.DestinationConfig) error {
+	if config.Mode != storages.BatchMode {
+		return nil
+	}
+
+	ctx := context.Background()
+
+	monitor := coordination.NewInMemoryService([]string{})
+	defer monitor.Close()
+
+	metaStorage, _ := meta.NewStorage(nil)
+	defer metaStorage.Close()
+
+	eventsCache := caching.NewEventsCache(metaStorage, 100)
+	defer eventsCache.Close()
+
+	factory := storages.NewFactory(ctx, "", monitor, eventsCache, nil, nil, metaStorage, 0)
+
+	randomValue := rand.Int()
+	testName := fmt.Sprintf("test_%v_%v", time.Now().Format(timestamp.DayLayout), randomValue)
+	config.DataLayout = &storages.DataLayout{
+		TableNameTemplate: testName,
+		UniqueIDField:     "id",
+	}
+	config.CachingConfiguration = &storages.CachingConfiguration{
+		Disabled: true,
+	}
+
+	storageProxy, _, err := factory.Create(testName, *config)
+	if err != nil {
+		logging.Errorf("[%s] Error initializing destination of type %s: %v", testName, config.Type, err)
+		return err
+	}
+
+	defer storageProxy.Close()
+
+	storage, err := storageProxy.Create()
```

Comment:
Instead of `Create()` func, storageProxy can have func like `GetStatus` or `GetError` which are return current error if a storage tried to initialize and got an error. Here we could wait initialization success or error and then do the rest of the logic.

Reply:
Connection testing is used in a process that was invoked by the HTTP handler.
In common cases, the HTTP calls have a timeout for waiting for an answer.
So in that case, there is no time for waiting while storage will be initialized after several minutes.
So I vote that in that place storage will be created immediately without any timeouts.